src/interfaces/controllers/insert-trailer-registration.ts

```diff
@@ -0,0 +1,54 @@
+import { HttpResponse } from 'aws-sdk';
+import { TrailerRegistration } from '../../domain/trailer-registration';
+import { ResponseHandler } from '../../utils/response-handler';
+import { DataAccess } from '../../utils/data-access';
+
+export class InsertTrailerRegistration {
+  public async createTrailer(trailerRegistration: TrailerRegistration): Promise<HttpResponse> {
+    const { vin, make } = trailerRegistration;
+    const vinOrChassisWithMake = vin.length === 17 ? vin : vin + make;
+
+    try {
+      const existingTrailerRegistration = await this.checkForExistingTrailerRegisteration(vinOrChassisWithMake);
+      // TODO: if exists then deregister the existing certificate
+      if (existingTrailerRegistration) {
+        this.deregisterTrailer(existingTrailerRegistration);
+      }
+    } catch (error) {
+      console.error(error);
+      ResponseHandler.failure(500, 'Internal Server Error');
+    }
+
+    // TODO: create new trailer registration
+
+    return ResponseHandler.success();
+  }
+
+  private async checkForExistingTrailerRegisteration(
+    vinOrChassisWithMake: string,
+  ): Promise<TrailerRegistration | null> {
+    // TODO: get data based on vinOrChassisWithMake
+    const data = (await DataAccess.getInstance().getById({
+      indexName: 'vinOrChassisWithMakeIndex',
+      keyCondition: '#vinOrChassisWithMake = :vinOrChassisWithMake',
+      expressionAttirbute: {
+        '#vinOrChassisWithMake': 'vinOrChassisWithMake',
+      },
+      expressionAttributeValues: {
+        ':vinOrChassisWithMake': vinOrChassisWithMake,
+      },
+    })) as { Count: number; Items: any };
+    if (!data.Count) {
+      return null;
+    }
```

Comment:
I'm not a fan of exposing the internals of our persistence layer outside of the DAO - isn't that what DAO's are for?  I'd expect the DAO to handle all this - and the DAO to expose a getByVinOrChassisWithMake method that deals with the internals. I'd also expect it to return a well-typed object.

Reply:
I was trying to create a more generic DataAccess which can be used as a reusable package but that can done as a separate activity. I've relocated the params now to the DataAccess layer.